code/controllers/controllers.js

```diff
@@ -59,4 +62,61 @@ exports.pushArticleVotes = (req, res, next) => {
     .catch((err) => {
       next(err);
     });
-};
\ No newline at end of file
+};
+
+exports.postComment = (req, res, next) => {
+  //Firstly, need to check of article exists before trying to insert a comment fo that article
+  selectOneArticle(req)
+    .then((result) => {
+      // Secondly, need to check user exists before trying to insert a comment
+      selectOneUser(req)
+        .then((result) => {
+          // now safe to insert comment because user exists
+          insertComment(req)
+            .then((result) => {
+              res.status(200).send({ comment: result });
+            })
+            .catch((err) => {
+              next(err);
+            });
+        })
+        .catch((err) => {
+              next(err);
+        });
+    })
+    .catch((err) => {
+              next(err);
+    });
+};
```

Comment:
This will prevent errors from actually being passed. This is because you have nested catches. It should be at the most outer point you can reach it.

Reply:
done - have now stripped out my bespoke error handling and put PSQL error handling in, so this whole block of code was reduced to just one call and catch. Also Emily spotted that I did not have 'return' in each of the calls so that is why I needed to have all those extra catch's.